src/java/htsjdk/samtools/util/SamLocusIterator.java

```diff
@@ -75,12 +77,15 @@ public RecordAndOffset(final SAMRecord record, final int offset) {
 
     /**
      * The unit of iteration.  Holds information about the locus (the SAMSequenceRecord and 1-based position
-     * on the reference), plus List of ReadAndOffset objects, one for each read that overlaps the locus
+     * on the reference), plus List of ReadAndOffset objects, one for each read that overlaps the locus;
+     * two more List of ReadAndOffset objects includes reads that overlaps the locus with insertions/deletions
      */
     public static final class LocusInfo implements Locus {
         private final SAMSequenceRecord referenceSequence;
         private final int position;
         private final List<RecordAndOffset> recordAndOffsets = new ArrayList<RecordAndOffset>(100);
+        private final List<RecordAndOffset> deletedInRecord = new ArrayList<RecordAndOffset>();
```

Comment:
This class is already quite slow due to all the object creation involved.  What do you think about creating these two ArrayList objects lazily at the first call to `addDeleted` and `addInsertion` since they may frequently be empty?